store/src/java/com/zimbra/cs/account/AuthTokenProperties.java

```diff
@@ -94,12 +93,13 @@ public AuthTokenProperties(Account acct, boolean isAdmin, Account adminAcct, lon
         tokenID = new Random().nextInt(Integer.MAX_VALUE-1) + 1;
     }
 
+    //Zimbra Mobile Gateway (zmgApp) has been removed from the product as part of ZBUG-3249, however this Java signature is needed to compile zm-mailbox
     public AuthTokenProperties(String acctId, boolean zmgApp, String externalEmail, String pass, String digest, long expires) {
         accountId = acctId;
         this.expires = expires;
-        externalUserEmail = externalEmail == null && !zmgApp ? "public" : externalEmail;
+        // commented out as part of ZBUG-3249: externalUserEmail = externalEmail == null && !zmgApp ? "public" : externalEmail;
```

Comment:
Isn't it enough to remove `!zmgApp` condition check only?
```suggestion
        externalUserEmail = externalEmail == null ? "public" : externalEmail;
```